osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/application/services/SecretManagementService.java

```diff
@@ -91,10 +102,19 @@ public SecretManagementService(RsaEncryptionProvider rsaEncryptionProvider,
     @Override
     public byte[] getDlmsGlobalUnicastEncryptionKey(String deviceIdentification) {
 
+        GetSecretsResponse response;
+
         try {
             GetSecretsRequest request = getSoapRequestForKey(deviceIdentification,
                     SecretType.E_METER_ENCRYPTION_KEY_UNICAST);
-            GetSecretsResponse response = secretManagementClient.getSecretsRequest(request);
+            response = secretManagementClient.getSecretsRequest(request);
+        }
+        catch(Exception e) {
+            throw new ConnectionException("Error while communicating with secret management "
```

Comment:
Same remarks as stated earlier: consider have more specific/informative error messages.

Reply:
We are in the process of finding out what goes wrong in production. This is a catch all

Reply:
'same issue as above'. No need to repeat and repeat.